atc/lidar/checker.go

```diff
@@ -47,6 +49,17 @@ func (c *checker) Run(ctx context.Context) error {
 	for _, ck := range checks {
 		if _, exists := c.running.LoadOrStore(ck.ID(), true); !exists {
 			go func(check db.Check) {
+				_, span := tracing.StartSpanFollowing(
```

Comment:
it got me wondering if we should have a span right there in `exec.CheckStep.Run` :thinking:  

that way, we could follow the same pattern that we have for all of the other steps, and also be able to capture errors (something we wouldn't be able to do so just from here)

https://github.com/concourse/concourse/blob/0b79feeb5ab7fa205c60ae6d34334152390cc74c/atc/exec/task_step.go#L125-L138

Reply:
i think this is how we did it on the first try, but found that the exec part of the trace pretty much overlapped completely with the lidar part..

Reply:
I see, that's true that this won't capture any errors within the check step. I think we could add it back but maybe I can split that out into a different issue and have a separate PR for it?